pshtt/pshtt.py

```diff
@@ -568,19 +576,22 @@ def https_check(endpoint):
     # remove the https:// from prefix for sslyze
     try:
         hostname = endpoint.url[8:]
-        server_tester = ServerConnectivityTester(hostname=hostname, port=443)
-        server_info = server_tester.perform()
+        server_location = ServerNetworkLocationViaDirectConnection.with_ip_address_lookup(
+            hostname=hostname, port=443
+        )
+        server_tester = ServerConnectivityTester()
+        server_info = server_tester.perform(server_location)
         endpoint.live = True
-        ip = server_info.ip_address
+        ip = server_location.ip_address
         if endpoint.ip is None:
             endpoint.ip = ip
         else:
             if endpoint.ip != ip:
                 utils.debug("{}: Endpoint IP is already {}, but requests IP is {}.".format(endpoint.url, endpoint.ip, ip))
-        if server_info.client_auth_requirement.name == 'REQUIRED':
+        if server_info.tls_probing_result.client_auth_requirement.name == 'REQUIRED':
             endpoint.https_client_auth_required = True
             logging.warning("{}: Client Authentication REQUIRED".format(endpoint.url))
-    except ServerConnectivityError as err:
+    except ConnectionToServerFailed as err:
         endpoint.live = False
         endpoint.https_valid = False
         logging.exception("{}: Error in sslyze server connectivity check when connecting to {}".format(endpoint.url, err.server_info.hostname))
```

Comment:
This has changed in `sslyze`:

```suggestion
        logging.exception("{}: Error in sslyze server connectivity check when connecting to {}".format(endpoint.url, err.server_location.hostname))
```